tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -813,17 +853,107 @@ task CalculateContamination {
     grep -v ^sample ~{base_name}_contamination.tsv | awk 'BEGIN{FS="\t"}{print($3)}' > contamination_error.txt
   >>>
 
-    runtime {
-      docker: docker
-      cpu: cpu
-      memory: "~{memory_mb} MiB"
-      disks: "local-disk ~{disk_size_gb} HDD"
-    }
+  runtime {
+    docker: docker
+    cpu: cpu
+    memory: "~{memory_mb} MiB"
+    disks: "local-disk ~{disk_size_gb} HDD"
+  }
 
-    output {
-        File pileups = "~{base_name}_pileups.tsv"
-        File contamination_table = "~{base_name}_contamination.tsv"
-        Float contamination = read_float("contamination.txt")
-        Float contamination_error = read_float("contamination_error.txt")
-    }
+  output {
+      File pileups = "~{base_name}_pileups.tsv"
+      File contamination_table = "~{base_name}_contamination.tsv"
+      Float contamination = read_float("contamination.txt")
+      Float contamination_error = read_float("contamination_error.txt")
+  }
+}
+
+task SamToFastq {
+  input {
+    File bam
+    String output_prefix
+    String docker = "us.gcr.io/broad-gotc-prod/picard-cloud:2.26.11"
+    Int memory_mb = 4096
+    Int disk_size_gb = 3*ceil(size(bam, "GiB")) + 128
+  }
+
+  command {
+    java -jar /usr/picard/picard.jar SamToFastq \
+    I=~{bam} \
+    FASTQ=~{output_prefix}_1.fastq.gz \
+    SECOND_END_FASTQ=~{output_prefix}_2.fastq.gz
+
+  }
+
+  runtime {
+    docker: docker
+    memory: "~{memory_mb} MiB"
+    disks: "local-disk ~{disk_size_gb} HDD"
+  }
+
+  output {
+    File fastq1 = output_prefix + "_1.fastq.gz"
+    File fastq2 = output_prefix + "_2.fastq.gz"
+  }
+}
+
+task FastQC {
+  input {
+    File unmapped_bam
+    Float? mem = 4
+    String docker = "us.gcr.io/tag-public/tag-tools:1.0.0" # sato: This likely needs to be made public
```

Comment:
This is a docker used by TAG team from GP; should I just ask for a docker file from them?

Reply:
if its something they regularly maintain and do security scans on then its okay to leave as is, otherwise we'll take care of bringing it to warp

Reply:
@takutosato pinging on this

Reply:
@wjdingman1 Sorry for the delay. It's updated about once every year and I doubt they do any security scans. I can share the docker file. It looks like the docker installs extraneous tools, it might be good to create a new docker (not necessarily in this PR)---could I get your help trimming it down?